core/os/thread.h

```diff
@@ -119,6 +119,8 @@ class Thread {
 public:
 	static void _set_platform_functions(const PlatformFunctions &p_functions);
 
+	_FORCE_INLINE_ static void yield() { std::this_thread::yield(); }
+
```

Comment:
I added the ability to specify a platform-specific thread yield, as `test_rid.h` was failing to compile on Apple, due to directly calling `std::this_thread::yield()`.